src/components/token/Tokens.js

```diff
@@ -20,20 +23,155 @@ class Tokens extends React.Component {
     constructor(props) {
         super(props);
 
+        /**
+         * tokens: List of tokens currently being rendered
+         * hasAfter: Indicates if a next page exists
+         * hasBefore: Indicates if a previous page exists
+         * searchText: Input text written by user
+         * sortBy: Which field to sort (uid, name, symbol)
+         * order: If sorted field must be ordered asc or desc
+         * page: Current page. Used to know if there is a previous page
+         * pageSearchAfter: searchAfter param needed to get the next page
+         * loading: Initial loading, when user clicks on the Tokens navigation item
+         * isSearchLoading: Indicates if search results are being retrieved from explorer-service
+         * calculatingPage: Indicates if next page is being retrieved from explorer-service
+         */
         this.state = {
             tokens: [],
             hasAfter: false,
             hasBefore: false,
-            searchText: ""
+            searchText: "",
+            sortBy: "uid",
+            order: "asc",
+            page: 1,
+            pageSearchAfter: [],
+            loading: true,
+            isSearchLoading: false,
+            calculatingPage: false
         }
     }
 
-    onSearchButtonClicked = () => {
-        // TODO: Implement behavior when search button is clicked
+    componentDidMount = async () => {
+        //"Click" on search to make the first query
+        await this.onSearchButtonClicked();
+        this.setState({ loading: false });
+    }
+
+    /**
+     * After the search button is clicked, this function resets the page and searchAfter params
+     */
+    resetPageSearchAfter = () => {
+        this.setState({
+            pageSearchAfter: [
+                {
+                    page: 1,
+                    searchAfter: []
+                }
+            ]
+        })
     }
 
+    /**
+     *
+     * Call explorer-service to get list of tokens according to the search criteria
+     *
+     * @param {*} searchAfter Parameter needed by ElasticSearch for pagination purposes
+     * @returns tokens
+     */
+    getTokens = async (searchAfter) => {
+        const tokensRequest = await tokensApi.getList(this.state.searchText, this.state.sortBy, this.state.order, searchAfter);
+        const tokens = get(tokensRequest, 'data', { hits: [], 'has_next': false });
+        tokens.hits = tokens.hits.map(token => ({ ...token, 'uid': token.id, 'nft': get(token, 'nft', false) }));
+        return tokens;
+    }
+
+    /**
+     * Process events when user clicks on search button
+     */
+    onSearchButtonClicked = async () => {
+        this.setState({ isSearchLoading: true });
+        //When search button is clicked, results return to the first page
+        this.resetPageSearchAfter();
```

Comment:
We only use this method here, since it only is a `setState`, why not incorporate it on the `setState` below?

Reply:
Will change on next commit